scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2336,6 +2341,10 @@ def __init__(
     def _build_kubelet_url(kubelet_url, host_ip, node_name):
         if node_name and host_ip:
             return kubelet_url.substitute(node_name=node_name, host_ip=host_ip)
+
+        global_log.warn(
```

Comment:
Technically, we could also throw here since in case this method returns ``None`` everything else will fail (and IIRC that method is called only once in the constructor so there is no chance for it evaluating to a different value later on).